lib/shared/bucketing/__tests__/bucketing.test.ts

```diff
@@ -781,6 +782,84 @@ describe('Config Parsing and Generating', () => {
         expect(c).toEqual(expected)
     })
 
+    it('buckets a user with user_id if no bucketingKey', () => {
+        const user = {
+            country: 'U S AND A',
+            user_id: 'pass_rollout',
+            customData: {
+                favouriteFood: 'pizza',
+            },
+            privateCustomData: {
+                favouriteDrink: 'coffee',
+            },
+            platformVersion: '1.1.2',
+            os: 'Android',
+            email: 'test@notemail.com',
+        }
+
+        const cWithBucketingKey = configWithBucketingKey('user_id')
+        const bucketedConfig = generateBucketedConfig({
+            config: config,
+            user,
+        })
+
+        const bucketedConfigFromBucketingKey = generateBucketedConfig({
+            config: cWithBucketingKey,
+            user,
+        })
+        expect(bucketedConfig).toEqual(bucketedConfigFromBucketingKey)
+    })
+
+    it('buckets a user with custom bucketingKey', () => {
+        const user = {
+            country: 'U S AND A',
+            user_id: 'pass_rollout',
+            customData: {
+                favouriteFood: 'pizza',
+            },
+            privateCustomData: {
+                favouriteDrink: 'coffee',
+            },
+            platformVersion: '1.1.2',
+            os: 'Android',
+            email: 'test@notemail.com',
+        }
+        const sameUserDifferentFood = {
+            ...user,
+            customData: {
+                favouriteFood: 'pasta',
+            },
+        }
+        const differentUserSameFood = {
+            ...sameUserDifferentFood,
+            user_id: 'a_different_person',
+        }
+        const configBucketedByFood = configWithBucketingKey('favouriteFood')
+        const bucketedConfigOrig = generateBucketedConfig({
+            config: config,
+            user,
+        })
+        const bucketedConfigByFood = generateBucketedConfig({
+            config: configBucketedByFood,
+            user,
+        })
+        const bucketedConfigSameUserDifferentFood = generateBucketedConfig({
+            config: configBucketedByFood,
+            user: sameUserDifferentFood,
+        })
+        const bucketedConfigDifferentUserSameFood = generateBucketedConfig({
+            config: configBucketedByFood,
+            user: differentUserSameFood,
+        })
+        expect(bucketedConfigOrig).not.toEqual(bucketedConfigByFood)
+        expect(bucketedConfigByFood).not.toEqual(
```

Comment:
might be worth double checking that these `not toEqual` assertions wont flake. 
It should be fine because I think the data is completely static, but in theory having different bucketing mechanisms doesn't guarantee that the configs will be different

Reply:
made them check a feature I added in the test data that has a bunch of different variations and check that they don't match (had to test a couple of times that they don't match, they should be consistently pass moving forward)

Reply:
coolio